core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -54,246 +47,34 @@ import org.apache.spark.sql.types.StructType
 import org.apache.spark.util.SerializableConfiguration
 import org.apache.spark.util.Utils
 
-/**
- * Records information about a checkpoint.
- *
- * This class provides the checksum validation logic, needed to ensure that content of
- * LAST_CHECKPOINT file points to a valid json. The readers might read some part from old file and
- * some part from the new file (if the file is read across multiple requests). In some rare
- * scenarios, the split read might produce a valid json and readers will be able to parse it and
- * convert it into a [[CheckpointMetaData]] object that contains invalid data. In order to prevent
- * using it, we do a checksum match on the read json to validate that it is consistent.
- *
- * For old Delta versions, which do not have checksum logic, we want to make sure that the old
- * fields (i.e. version, size, parts) are together in the beginning of last_checkpoint json. All
- * these fields together are less than 50 bytes, so even in split read scenario, we want to make
- * sure that old delta readers which do not do have checksum validation logic, gets all 3 fields
- * from one read request. For this reason, we use `JsonPropertyOrder` to force them in the beginning
- * together.
- *
- * @param version the version of this checkpoint
- * @param size the number of actions in the checkpoint, -1 if the information is unavailable.
- * @param parts the number of parts when the checkpoint has multiple parts. None if this is a
- *              singular checkpoint
- * @param sizeInBytes the number of bytes of the checkpoint
- * @param numOfAddFiles the number of AddFile actions in the checkpoint
- * @param checkpointSchema the schema of the underlying checkpoint files
- * @param checksum the checksum of the [[CheckpointMetaData]].
- */
-@JsonPropertyOrder(Array("version", "size", "parts"))
-case class CheckpointMetaData(
```

Comment:
Class moved to its own file, right? 
Any functional changes?

Reply:
No functional changes.